File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/AppSettingsComponent.java
package io.github.huacnlee.autocorrectIdeaPlugin;

import com.intellij.ui.components.JBBox;
import com.intellij.ui.components.JBCheckBox;
import com.intellij.util.ui.FormBuilder;

import javax.swing.*;

public class AppSettingsComponent {
    private final JPanel myMainPanel;
    private final JBCheckBox enableStatus = new JBCheckBox("Enable", true);
    private final JBCheckBox enableFormatOnSaveStatus = new JBCheckBox("Format On Save", true);
    private final JBCheckBox enableLintStatus = new JBCheckBox("Enable Lint", true);

    public AppSettingsComponent() {

        myMainPanel = FormBuilder.createFormBuilder()
                .addComponent(enableFormatOnSaveStatus, 1)
                .addComponent(new JLabel("Turn on/off autoformatting file on save."))
                .addComponent(enableLintStatus, 1)
                .addComponent(new JLabel("Turn on/off AutoCorrect lint."))
                .addComponentFillVertically(new JPanel(), 0)
                .getPanel();
    }

    public JPanel getPanel() {
        return myMainPanel;
    }

    public JComponent getPreferredFocusedComponent() {
        return enableStatus;
    }

    public boolean getEnable() {
        return enableStatus.isSelected();
    }

    public void setEnable(boolean newStatus) {
        enableStatus.setSelected(newStatus);
    }

    public boolean getEnableFormatOnSave() {
        return enableFormatOnSaveStatus.isSelected();
    }

    public void setEnableFormatOnSave(boolean newStatus) {
        enableFormatOnSaveStatus.setSelected(newStatus);
    }

    public boolean getEnableLint() {
        return enableLintStatus.isSelected();
    }

    public void setEnableLint(boolean newStatus) {
        enableLintStatus.setSelected(newStatus);
    }

}

File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/AppSettingsComponent.kt
package io.github.huacnlee.autocorrectIdeaPlugin

import com.intellij.ui.components.JBCheckBox
import com.intellij.util.ui.FormBuilder
import javax.swing.JComponent
import javax.swing.JLabel
import javax.swing.JPanel

class AppSettingsComponent {
    val panel: JPanel
    private val enableStatus = JBCheckBox("Enable", true)
    private val enableFormatOnSaveStatus = JBCheckBox("Format On Save", true)

    init {
        panel = FormBuilder.createFormBuilder() //                .addComponent(enableStatus, 1)
            //                .addComponent(new JLabel("Turn on/off AutoCorrect."))
            //                .addVerticalGap(2)
            //                .addSeparator()
            //                .addVerticalGap(2)
            .addComponent(enableFormatOnSaveStatus, 1)
            .addComponent(JLabel("Turn on/off autoformatting file on save."))
            .addComponentFillVertically(JPanel(), 0)
            .panel
    }

    val preferredFocusedComponent: JComponent
        get() = enableStatus
    var enable: Boolean
        get() = enableStatus.isSelected
        set(newStatus) {
            enableStatus.isSelected = newStatus
        }
    var enableFormatOnSave: Boolean
        get() = enableFormatOnSaveStatus.isSelected
        set(newStatus) {
            enableFormatOnSaveStatus.isSelected = newStatus
        }
}